### ISSUE TITLE
Worker stop handle jobs if redis is disconnected

### ISSUE THREAD

**Issue by AzHicham** [geofmureithi/apalis#49]
Hello in some case as follow a worker may stop listening for new tasks, this happens when the connection between the worker & redis  is interrupted. 

Start a worker : 
```
pub async fn send_email(job: Email, _ctx: JobContext) -> impl IntoJobResponse {
    println!("sleeping");
    let dur = time::Duration::from_millis(5_000);
    thread::sleep(dur);
    println!("job number {}", job.text);
}

#[tokio::main]
async fn main() -> Result<()> {
    let storage = RedisStorage::connect("redis://127.0.0.1/").await.expect("");
    Monitor::new()
        .register(
            WorkerBuilder::new(storage.clone())
                .build_fn(send_email),
        )
        .run()
        .await
}
```
Then with a client send 20 tasks for example.

Once some job are processed, restart redis, under macOS: `brew service restart redis`. 
If the worker was already processing a job (eg in our example sleeping ...) 
then the worker will finish this task and will NOT handle the rest of 20 initial tasks. 

But you can start another worker in parallel and it will handle the reste of the task without any issue.

If this issue can be fixed this will allow enable the Try Reconnect feature in the middle of application and also at startup (celery style)

**Comment by AzHicham** [comment]
Hello @geofmureithi 
Did you get time to investigate this issue ? 
Since this could be really problematic ^^

**Comment by geofmureithi** [comment]
Hey @AzHicham This can be fixed by using https://docs.rs/redis/latest/redis/aio/struct.ConnectionManager.html in Redis storage. It will be fixed in the next release.

**Comment by AzHicham** [comment]
Ok nice to know !!! Thank you

**Comment by shan-pandian** [comment]
Hi @geofmureithi, Thanks for providing this amazing repo. This one really suits our need. We are currently trying it out. We are on 0.3.6 and considering moving to 0.4.1. We had this issue hit us couple of times and we need to restart our services to start processing the jobs. 

Is this issue still open? would be great to know what version we can expect this fix?

**Comment by geofmureithi** [comment]
Hey. Am aware of this issue. The goal was to move from a concrete connection manager to using the generic one provided by the redis crate.
Unfortunately I was not able to do that in 0.4. I don't want to change the API with a new version release. For now I will do a hotfix and do a release that will sort this issue then tackle the API issue later.

**Comment by geofmureithi** [comment]
@AzHicham @shan-pandian Please verify this is fixed in v0.4.2 if possible.
